src/main/java/com/redhat/labs/lodestar/model/Engagement.java

```diff
@@ -97,7 +98,48 @@
     @JsonbProperty("commit_message")
     private String commitMessage;
 
-    @JsonbTransient
-    private FileAction action;
+    public static Engagement deepCopy(Engagement engagement) {
```

Comment:
Just wondering about this method and ability to maintain with data additons. Did you look at something like https://www.baeldung.com/java-deep-copy#3-json-serialization-with-jackson

Reply:
you are correct.  the json serialization is a simpler and more maintainable solution.  i have updated the code to use this instead of the original deep copy method.  the ObjectMapper and new `clone` method can be found in EngagementService.java.